internal/core/layout.rs

```diff
@@ -78,6 +78,33 @@ impl core::ops::Add for LayoutInfo {
     }
 }
 
+/// Returns the logical min and max sizes given the provided layout constraints.
+pub fn min_max_size_for_layout_constraints(
+    constraints_horizontal: LayoutInfo,
+    constraints_vertical: LayoutInfo,
+) -> (Option<crate::lengths::LogicalSize>, Option<crate::lengths::LogicalSize>) {
+    let min_width = constraints_horizontal.min.min(constraints_horizontal.max) as f32;
+    let min_height = constraints_vertical.min.min(constraints_vertical.max) as f32;
+    let max_width = constraints_horizontal.max.max(constraints_horizontal.min) as f32;
+    let max_height = constraints_vertical.max.max(constraints_vertical.min) as f32;
+
+    let min_size = if min_width > 0. || min_height > 0. {
+        Some(crate::lengths::LogicalSize::new(min_width, min_height))
+    } else {
+        None
+    };
+    let max_size = if max_width > 0.
+        && max_height > 0.
```

Comment:
In fact, if the maximum size is 0, with the current code, you simply do no longer set any maximum size. 
This allows the user to bypass the maximum size entirely.  I wonder if we should not just put a threshold size in that case. (like very small but not 0)

Reply:
I'll merge the change (to fix the panic), but I'm definitely open to discussing this aspect further. I'm curious how this would look like?

Do you mean that if the max size is 0, we should use a threshold like the global strut in Qt, say 100x100?